apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/transaction/DexOfferTransaction.java

```diff
@@ -84,41 +85,77 @@ public void validateAttachment(Transaction transaction) throws AplException.Vali
             throw new AplException.NotCurrentlyValidException(JSON.toString(incorrect("amountOfTime",  String.format("value %d not in range [%d-%d]", attachment.getFinishTime(), 0, MAX_ORDER_DURATION_SEC))));
         }
 
+        if (shouldFreezeAPL(attachment.getType(), attachment.getOfferCurrency())) {
+            Long fee = transaction.getFeeATM();
+            Long amountATM = attachment.getOfferAmount();
+            long totalAmountATM = Math.addExact(amountATM, fee);
+            Account sender = Account.getAccount(transaction.getSenderId());
+
+            if (sender.getUnconfirmedBalanceATM() < totalAmountATM) {
+                throw new AplException.NotCurrentlyValidException("Not enough money.");
+            }
+        }
+
+
     }
 
+
     @Override
     public boolean applyAttachmentUnconfirmed(Transaction transaction, Account senderAccount) {
```

Comment:
Maybe move 'freeze' logic to this method to prevent double-spending before confirmation?